src/signals/models.py

```diff
@@ -221,10 +247,12 @@ class Signal(models.Model):
         help_text=_('Source Subdivision'),
         on_delete=models.PROTECT,
     )
-
-    @property
-    def base(self) -> str:
-        return self.source.reference_signal
+    base = models.ForeignKey(
+        'signals.SignalGroup',
+        related_name='signals',
+        help_text=_('Base'),
+        on_delete=models.PROTECT,
+    )
```

Comment:
hmm, this might make getting properties of the base signal (like its description text) awkward, since you'd have to search `base.signals` for a signal matching `base.name`

why not store the base signal on the signal group as an fk? then this becomes

```suggestion
    group = models.ForeignKey(
        'signals.SignalGroup',
        related_name='signals',
        help_text=_('Group'),
        on_delete=models.PROTECT,
    )

    @property
    def base(self) -> str:
        return self.group.base
```